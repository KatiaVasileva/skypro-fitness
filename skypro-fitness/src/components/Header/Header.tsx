function Header() {
  const isAuth = false;

  return (
    <div className="container mx-auto">
      <div className="h-50 flex flex-row items-center place-content-between mb-3">
        <img src="/img/logo.png" className="w-55" />

        {!isAuth && (
          <button className="btn-primary md:w-26 md:h-13">
            Войти
          </button>
        )}

        {isAuth && (
          <div className="flex flex-row items-center w-42 gap-3">
            <img src="/img/profile.png" className="w-12 h-12"></img>
            <p className="text-2xl font-normal text-black">Сергей</p>
            <img src="/img/rectangle.png" className="w-4 h-3 "></img>
          </div>
        )}

      </div>
      <p className="invisible md:visible md:text-lg/5 md:font-normal  md:text-gray ">
        Онлайн-тренировки для занятий дома
      </p>
    </div>
  );
}

export default Header;
