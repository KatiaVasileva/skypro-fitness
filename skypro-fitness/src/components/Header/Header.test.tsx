/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Header from "./Header";
import * as router from "react-router";
import UserProvider from "../../context/UserContext";
import CoursesProvider from "../../context/CoursesContext";

const navigate = jest.fn();

describe("Header component", () => {
  beforeEach(() => {
    jest.spyOn(router, "useNavigate").mockImplementation(() => navigate);
  });

  it("renders correctly", () => {
    const { container } = render(
      <UserProvider>
        <CoursesProvider>
          <Header />
        </CoursesProvider>
      </UserProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
