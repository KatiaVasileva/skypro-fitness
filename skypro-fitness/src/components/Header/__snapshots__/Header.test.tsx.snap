// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="h-50 flex flex-row items-center place-content-between mb-3"
    >
      <img
        class="h-[35px] w-[220px] cursor-pointer"
        src="/img/logo.png"
      />
      <button
        class="btn-primary md:w-26 h-9 md:h-13"
      >
        Войти
      </button>
    </div>
    <p
      class="hidden md:block md:text-lg/5 md:font-normal  md:text-gray "
    >
      Онлайн-тренировки для занятий дома
    </p>
  </div>
</div>
`;
